src/operation/union/OverlapUnion.cpp

```diff
@@ -0,0 +1,267 @@
+/**********************************************************************
+ *
+ * GEOS - Geometry Engine Open Source
+ * http://geos.osgeo.org
+ *
+ * Copyright (C) 2019 Daniel Baston <dbaston@gmail.com>
+ *
+ * This is free software; you can redistribute and/or modify it under
+ * the terms of the GNU Lesser General Public Licence as published
+ * by the Free Software Foundation.
+ * See the COPYING file for more information.
+ *
+ **********************************************************************/
+
+#include <geos/operation/union/OverlapUnion.h>
+
+#include <geos/geom/Coordinate.h>
+#include <geos/geom/CoordinateSequence.h>
+#include <geos/geom/CoordinateSequenceFilter.h>
+#include <geos/geom/Envelope.h>
+#include <geos/geom/GeometryFactory.h>
+#include <geos/geom/LineSegment.h>
+#include <geos/geom/util/GeometryCombiner.h>
+#include <geos/util/TopologyException.h>
+
+namespace geos {
+namespace operation {
+namespace geounion {
+
+// https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/operation/union/OverlapUnion.java
+
+// xxx OverlapUnion::union leaks like a seive, check cascadedunion implementation for use of unique_ptr
+
+using namespace geom;
+using namespace geom::util;
+
+/* public */
+Geometry*
+OverlapUnion::doUnion()
+{
+    Envelope overlapEnv = overlapEnvelope(g0, g1);
+    /**
+     * If no overlap, can just combine the geometries
+     */
+    if (overlapEnv.isNull()) {
+        // Geometry* g0Copy = g0->clone().get();
+        // Geometry* g1Copy = g1->clone().get();
+        return GeometryCombiner::combine(g0, g1).release();
+    }
+
+    std::vector<Geometry*> disjointPolys;
+
+    std::unique_ptr<Geometry> g0Overlap = extractByEnvelope(overlapEnv, g0, disjointPolys);
+    std::unique_ptr<Geometry> g1Overlap = extractByEnvelope(overlapEnv, g1, disjointPolys);
+
+    // std::out << "# geoms in common: " << intersectingPolys.size() << std::endl;
+    std::unique_ptr<Geometry> theUnion(unionFull(g0Overlap.get(), g1Overlap.get()));
```

Comment:
If this throws, `disjointPolys` will be leaked, no?

Reply:
Pretty sure they leak no matter what 😂

Reply:
disjointPolys now managed